internal/fsapi/file.go

```diff
@@ -380,3 +375,158 @@ type File interface {
 	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/close.html
 	Close() syscall.Errno
 }
+
+// Readdir is the result of File.Readdir and it is a cursor over a directory.
+//
+// # Errors
+//
+// All methods that can return an error return a syscall.Errno, which is zero
+// on success.
+//
+// # Notes
+//
+//   - Implementations are allowed to keep a buffer or a sliding window of values
+//     over a directory. Because reading in batches requires to hold a reference
+//     to an open directory, callers should pay extra care to make sure
+//     that such references are released, by invoking Close.
+//   - This is especially important on the Windows platform, where keeping
+//     a file handle open may prevent concurrent processes from accessing
+//     (e.g. deleting) the underlying directory. This may conflict with cleanup
+//     procedures (such as `testing/TB.Cleanup()`).
+type Readdir interface {
+	// Skip is equivalent to calling Next n times.
+	Skip(n uint64)
+
+	// Offset returns the offset of the current value under the internal cursor.
+	Offset() uint64
+
+	// Reset seeks the internal cursor to 0 and refills the buffer.
+	//
+	// # Errors
+	//
+	// A zero syscall.Errno is success. The below are expected otherwise:
+	//   - syscall.EINVAL: the offset is larger than the current internal cursor.
+	//   - syscall.EBADF: the file or directory was closed.
+	//
+	// # Notes
+	//
+	//   - This is conceptually similar to invoking Seek(0, io.SeekStart) on io.Seeker
+	//     and `fseek` in POSIX over an open directory. See
+	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/fseek.html
+	//   - However implementations will generally reopen the underlying directory
+	//     for portability reasons.
+	Reset() syscall.Errno
+
+	// Rewind seeks the internal cursor to the given offset.
+	//
+	// Rewinding to offset 0 usually should produce the same result
```

Comment:
I would love to have only one function. Can you give me the summary of what would happen if we deleted `Reset`?

Reply:
just a useless check for offset == 0 every time you call it, but it's ok. Will do!